template.yaml

```diff
@@ -567,67 +564,62 @@ Resources:
           PredefinedMetricType: ECSServiceAverageCPUUtilization
         TargetValue: 70.0
 
-  ## www domain
-
-  #  FrontCustomDomain:
-  #    Type: AWS::ApiGatewayV2::DomainName
-  #    Properties:
-  #      DomainName: !FindInMap [ EnvironmentVariables, !Ref Environment, FEURL ]
-  #      DomainNameConfigurations:
-  #        - CertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
-  #          EndpointType: REGIONAL
-  #          SecurityPolicy: TLS_1_2
-
-  #  FrontApiRecord:
-  #    Type: AWS::Route53::RecordSet
-  #    Properties:
-  #      Name: !FindInMap [EnvironmentVariables, !Ref Environment, FEURL ]
-  #      Type: A
-  #      HostedZoneId: !Sub "{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}"
-  #      AliasTarget:
-  #        DNSName: !GetAtt FrontCustomDomain.RegionalDomainName
-  #        HostedZoneId: !GetAtt FrontCustomDomain.RegionalHostedZoneId
-  #        EvaluateTargetHealth: false
-
-  #  FrontApiMapping:
-  #    Type: AWS::ApiGatewayV2::ApiMapping
-  #    Properties:
-  #      DomainName: !FindInMap [EnvironmentVariables, !Ref Environment, FEURL ]
-  #      ApiId: !Ref ApiGwHttpEndpoint
-  #      Stage: "$default"
-  #    DependsOn:
-  #      - FrontCustomDomain
+### Front End API Gateway Custom Domain definition
 
+  F2FFrontCustomDomain:
+    Type: AWS::ApiGateway::DomainName
+    Properties:
+      DomainName: !If
+        - IsNotDevelopment
+        - !Sub
+          - "www.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+        - !Sub
+          - "${AWS::StackName}.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [EnvironmentVariables, !Ref Environment, DNSSUFFIX]
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
+      SecurityPolicy: TLS_1_2
+    DependsOn:
+      - ApiGwHttpEndpoint
+      - APIStageDefault
```

Comment:
this dependency block should be in the BasePathMapping resource (which is implicitly dependent on this DomainName resource)